### PR TITLE
sharing with user and groups with different case username and groupname

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

Add webui acceptance tests to:

- share with users and using different case of the username
- share with groups using different case of the group name

## Related Issue

Part of https://github.com/owncloud/core/issues/35354

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 1]
```diff
@@ -213,4 +213,45 @@ Feature: Sharing files and folders with internal groups
...
+    And user "user1" has been added to group "<group_id1>"
+    And user "user2" has been added to group "<group_id2>"
+    And user "user3" has been added to group "<group_id3>"
+    And user "some-user" has created folder "/simple-folder"
+    And user "some-user" has logged in using the webUI
+    When the user shares folder "simple-folder" with group "<group_id1>" using the webUI
+    And the user shares folder "simple-folder" with group "<group_id2>" using the webUI
+    And the user shares folder "simple-folder" with group "<group_id3>" using the webUI
+    And the user re-logs in as "user1" using the webUI
+    And folder "simple-folder" should be marked as shared with "<group_id1>" by "Regular User" on the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]:
```suggestion
    Then folder "simple-folder" should be marked as shared with "<group_id1>" by "Regular User" on the webUI
```

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 2]
```diff
@@ -213,4 +213,45 @@ Feature: Sharing files and folders with internal groups
...
+    And the user re-logs in as "user2" using the webUI
+    And folder "simple-folder" should be marked as shared with "<group_id2>" by "Regular User" on the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]:
```suggestion
    Then folder "simple-folder" should be marked as shared with "<group_id2>" by "Regular User" on the webUI
```

File: tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature [hunk 3]
```diff
@@ -213,4 +213,45 @@ Feature: Sharing files and folders with internal groups
...
+    And the user re-logs in as "user3" using the webUI
+    And folder "simple-folder (2)" should be marked as shared with "<group_id3>" by "Regular User" on the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalGroups/shareWithGroups.feature]:
```suggestion
    Then folder "simple-folder (2)" should be marked as shared with "<group_id3>" by "Regular User" on the webUI
```

IMO for this sort of test we can write `Then` 3 times.

File: tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature [hunk 4]
```diff
@@ -370,4 +370,25 @@ Feature: Sharing files and folders with internal users
...
+      | username       |
+      | user1          |
+      | brand-new-user |
+    And user "user1" has created folder "/simple-folder"
+    And user "user1" has shared folder "simple-folder" with user "brand-new-user"
+    And user "user1" has logged in using the webUI
+    And the user has opened the share dialog for folder "simple-folder"
+    When the user types "<sharee_id>" in the share-with-field
+    Then a tooltip with the text "No users or groups found for <sharee_id>" should be shown near the share-with-field on the webUI
+    When the user types "<sharee_id>" in the share-with-field
```
Comment (phil-davis) [inline tests/acceptance/features/webUISharingInternalUsers/shareWithUsers.feature]: What was intended here?
These steps will repeat typing the same `sharee_id` 3 times.
(and the whole scenario runs again for each of the 3 examples)

**Review by paurakhsharma — COMMENTED** [review]
Looks good to me :+1:

**Review by phil-davis — APPROVED** [review]

**Comment by HariBhandari07** [comment]
Backport on https://github.com/owncloud/core/pull/35651

**Comment by phil-davis** [comment]
This got today's annoying PHPunit drone random errors.
I restarted the drone build.

**Pull request merged**